src/lib.rs

```diff
@@ -6,6 +6,7 @@ pub mod error;
 pub mod field;
 pub mod helpers;
 pub mod net;
+pub mod pipeline;
```

Comment:
Alex suggested that we keep the toy project as test-only.  I think that this is the place where you would do that.

```suggestion
#[cfg(test)]
pub mod pipeline;
```